src/csharp/Microsoft.Spark.Worker/Command/SqlCommandExecutor.cs

```diff
@@ -524,4 +528,168 @@ public IArrowArray[] Run(ReadOnlyMemory<IArrowArray> input)
             }
         }
     }
+
+    internal class ArrowGroupedMapCommandExecutor : ArrowSqlCommandExecutor
+    {
+        protected override CommandExecutorStat ExecuteCore(
```

Comment:
The `ExecuteCore` methods will get simpler once we remove the `tmp` memory stream. So maybe we can revisit after that happens.

I would have hoped I could reuse the "CommandRunner" classes as well, but can't since the signature is different.